integrations/gtag/lib/index.js

```diff
@@ -725,43 +721,62 @@ function trackEnhancedEvent(eventName, payload) {
  * Set custom dimensions and metrics.
  *
  * @api private
- * @param options
- */
-
-function setCustomDimensionsAndMetrics(options) {
-  var gaOptions = options.gaOptions || {};
-  if (gaOptions && Object.keys(gaOptions).length) {
-    if (gaOptions.setAllMappedProps) {
-      // set custom dimension and metrics if present
-      // REF: https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets
-      // For content grouping
-      // https://support.google.com/analytics/answer/7475939?hl=en#code
-      var customMap = merge(gaOptions.dimensions, gaOptions.metrics);
-      if (options.gaWebMeasurementId) {
-        push(
-          'config',
-          options.gaWebMeasurementId,
-          merge(
-            {
-              custom_map: customMap
-            },
-            gaOptions.contentGroupings
-          )
-        );
+ * @param {Object} props
+ * @param {Object} options
+ */
+
+function setCustomDimensionsAndMetrics(props, options) {
+  if (options.gaSetAllMappedProps) {
+    // set custom dimension and metrics if present
+    // REF: https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets
+    // For content grouping
+    // https://support.google.com/analytics/answer/7475939?hl=en#code
+    var customMap = merge(
+      options.gaCustomDimensions || {},
+      options.gaCustomMetrics || {}
+    );
+    var customMapValues = {};
+    for (var field in customMap) {
+      if (customMap.hasOwnProperty(field)) {
+        customMapValues[customMap[field]] = props[field];
       }
-      if (options.gaWebAppMeasurementId) {
-        push(
-          'config',
-          options.gaWebAppMeasurementId,
-          merge(
-            {
-              custom_map: customMap
-            },
-            gaOptions.contentGroupings
-          )
-        );
+    }
```

Comment:
The section was changed for two reasons:
1. For parity with the existing GA destination.
2. The property names were being mapped, not the actual values.